tests/library/client-certificates.spec.ts

```diff
@@ -546,6 +546,7 @@ test.describe('browser', () => {
       keyPath: asset('client-certificates/client/trusted/key.pem'),
     };
     const page = await browser.newPage({
+      ignoreHTTPSErrors: true,
```

Comment:
Why this change? Seems unrelated.

Reply:
Before we were overriding the `ignoreHTTPSErrors` field when creating the socks proxy. Later on we were reading it in the APIRequestContext when making a http request, there we were readying the updated value rather than the original value. This is e.g. what this PR is fixing, that things like this won't happen anymore.